### PR TITLE
[Truffle] Implementing more Array#multiply.

### PR DESCRIPTION
**Pull request by bjfish**

### CODE REVIEW

File: truffle/src/main/java/org/jruby/truffle/nodes/core/ArrayNodes.java [hunk 1]
```diff
@@ -242,7 +269,47 @@ public RubyArray mulObject(RubyArray array, int count) {
...
+        }
+
+        @Specialization(guards = {"!isRubyString(arguments[1])"})
+        public Object mulObjectCount(VirtualFrame frame, RubyArray array, Object object ) {
+            notDesignedForCompilation();
+            if (respondToToStrNode == null) {
+                CompilerDirectives.transferToInterpreter();
+                respondToToStrNode = insert(KernelNodesFactory.RespondToNodeFactory.create(getContext(), getSourceSection(), new RubyNode[] { null, null, null }));
+            }
+            if(respondToToStrNode.doesRespondTo(frame, object, getContext().makeString("to_str"), false)){
```
Comment (chrisseaton) [inline truffle/src/main/java/org/jruby/truffle/nodes/core/ArrayNodes.java]: Given this is `notDesignedForCompilation`, you could even write the condition here using `ruby` - `if ((boolean) ruby(frame, "object.respond_to 'to_str'", "object", object)) {`. The avoids the need for the node. No need to change it - just thought I'd point it out.

File: truffle/src/main/java/org/jruby/truffle/nodes/core/ArrayNodes.java [hunk 2]
```diff
@@ -242,7 +269,47 @@ public RubyArray mulObject(RubyArray array, int count) {
...
+                return ruby(frame, "join(sep.to_str)", "sep", object);
+            } else {
+                if (toIntNode == null) {
+                    CompilerDirectives.transferToInterpreter();
+                    toIntNode = insert(ToIntNodeFactory.create(getContext(), getSourceSection(), null));
+                }
+                final int count = toIntNode.executeIntegerFixnum(frame, object);
+                if(count < 0){
```
Comment (chrisseaton) [inline truffle/src/main/java/org/jruby/truffle/nodes/core/ArrayNodes.java]: Spacing here and other places!

**Comment by chrisseaton** [comment]
Looks good - please try on the spacing though. Feel free to merge it yourself.

**Comment by bjfish** [comment]
@chrisseaton thanks, I think i've got my code formatter configured correctly now for spacing.

**Pull request merged**